Code/Java/CircularQueue

```diff
@@ -0,0 +1,360 @@
+        {
+            last++;
+        }
+        cq[last]=value;
+    }
+
+    public int delete()
+    {
+        // This function is used to delete an element from the Circular Queue. 
+        // In a circular queue, the element is always deleted from first position. 
+         
+        if(first == -1)
+        {
+            System.out.println("Underflow"); return 0; 
+        }
+        else
+        {
+            int temp = cq[first];
+            if(first == last)
+            { 
+                reset(); 
+            }
+            else if(first == cq.length-1)
+            {
+                first=0;
+            }
+            else
+            {
+                first++;
+            }
+            return temp;
+        }
+    }
+
+    private void display()
+    {
+        // This function is used to display a Circular Queue.
+         
+        if(first == -1)
+        {
+            System.out.print("Circular Queue is Empty");
+            return;
+        }
+        
+        for(int i = first; ; i++)
+        {
+            if(i == cq.length)
+            {
+                i=0;
+            }
+
+            System.out.print(cq[i] + " ");
+
+            if(i == last)
+            {
+                break;
+            }
+        }
+        System.out.println();
+    }
+
+    void reset()
+    {
+        // This function is used to reset first and last to original values -1.
+        first=last=-1;
+    }
+
+    void main()
+    {
+        // This function serves as a driver code.
+         
+        Scanner scanner = new Scanner(System.in);
+
+        System.out.println("Enter number of elements: ");
+        int size = scanner.nextInt(); scanner.nextLine();
+
+        CircularQueue ob = new CircularQueue(size);
+
+        int flag=1;
+
+        while(flag == 1)
+        {
+            System.out.println("Choose from following: ");
+            System.out.println("1. Insert an element");
+            System.out.println("2. Delete an element");
+            System.out.println("3. Display the elements: ");
+            System.out.println("Enter choice: ");
+            int choice = scanner.nextInt(); scanner.nextLine();
+
+            switch(choice)
+            {
+                case 1:
+                System.out.println("Enter element: ");
+                int value = scanner.nextInt(); scanner.nextLine();
+                ob.insert(value);
+                break;
+
+                case 2:
+                if(ob.delete()!=0)
+                { 
+                    System.out.println("Deleted element: "+ob.delete());
+                }
+                break;
+
+                case 3:
+                System.out.println("Displaying the queue");
+                ob.display();
+                break;
+            }
+
+            System.out.println("Do you wish to continue? Enter 1 for YES or 0 for NO: ");
+            flag = scanner.nextInt(); scanner.nextLine();
+        }
+    }
+}
+
+/*
+    Test Case:
+    
```

Comment:
Follow
Test Cases
Input:
Output:

Reply:
Since there is a repeated pattern of interspersed inputs and outputs, I am unable to make this change. Please suggest how I can better this. Thanks!

Reply:
Ohk, then lets leave it like this :)